applications/tari_merge_mining_proxy/src/main.rs

```diff
@@ -742,38 +831,53 @@ fn handle_connection(
             )
             .as_bytes()
             .to_vec();
-            match base_curl(json.len() as u64, &url, true, config) {
-                Ok(mut curl) => {
-                    match do_curl(&mut curl, &json) {
-                        // request_id
-                        Ok(response) => {
-                            data = response;
-                            debug!(target: LOG_TARGET, "Handling POST Method: {}", method);
-                            debug!(
-                                target: LOG_TARGET,
-                                "Response: {:#}",
-                                String::from_utf8_lossy(&data).to_string()
-                            );
-                            match handle_post(&method, &json, data.clone(), transient, consensus, rt, config) {
-                                Ok(result) => {
-                                    data = result;
+            if let "submitblock" = method.as_str() {
+                if transient.tari_difficulty.unwrap_or_default() < transient.monero_difficulty.unwrap_or_default() {
```

Comment:
I think it would be better to calculate the difficulty of the block here and then have two if statements:
```
if diff > monero_diff {
  submit_to_monero();
}
if diff > tari_diff {
  submit_to_tari();
}
```
At some point these should be done in parallel in order to minimise the time to submit and hence increase the chance of winning the reward. Having the two if statements also handles the case where `tari_difficulty` is higher than `monero_difficulty`, which is missing